jekyll/_cci2/workflows.md

```diff
@@ -0,0 +1,205 @@
+ 
+The Workflows feature also enables you to limit the job execution to a branch with the `filters` key. See the Branch-Level Job Execution section for instructions.
+ 
+## Running a Workflow from a Failed Job
+
+It is possible to rerun a job that failed in the middle of a workflow and continue the rest of the workflow using the Rerun button on the Workflows page of the CircleCI application.
+
+![Rerunning a Workflow from a Failed Job]({{ site.baseurl }}/assets/img/docs/restart_from_failed.png) 
+
+To rerun a workflow from the failed job:
+
+1. From the Builds page, select the project name. 
+2. Click the Workflows tab to display workflows for your project.  
+3. Select a workflow from the list to see all jobs in the workflow. 
+4. Click Rerun and select from failed to rerun the job and continue the workflow. 
+
+## Fan-Out/Fan-In Workflow Example
+	
+The illustrated example workflow runs a common build Job, then fans-out to run a set of acceptance test Jobs in parallel, and finally fans-in to run a common deploy Job.
+
+![Fan-out and Fan-in Workflow]({{ site.baseurl }}/assets/img/docs/fan_in_out.png) 
+
+The following `config.yml` snippet is an example of a workflow configured for sequential job execution:
+
+``` 
+workflows:
+  version: 2
+  build_accept_deploy:
+    jobs:
+      - build:
+      - acceptance_test_1:
+          requires:
+            - build
+      - acceptance_test_2:
+          requires:
+            - build
+      - acceptance_test_3:
+          requires:
+            - build
+      - acceptance_test_4:
+          requires:
+            - build          
+      - deploy:
+          requires:
+            - acceptance_test_1
+            - acceptance_test_2
+            - acceptance_test_3
+            - acceptance_test_4
+```
+ 
+## Branch-Level Job Execution
+The following example shows a workflow configured with jobs on three branches: Dev, Stage, and Pre-Prod. 
+
+![Branch-Level Job Execution]({{ site.baseurl }}/assets/img/docs/branch_level.png) 
+
+The following `config.yml` snippet is an example of a workflow configured for branch-level job execution:
+
+```
+workflows:
+  version:2
+  dev_stage_pre-prod_master_deploy:
+    jobs:
+      - test_dev:
+          filters:
+            branches:
+              only: dev
+      - test_stage:
+          filters:
+            branches:
+              only: stage
+      - test_pre-prod:
+          filters:
+            branches:
+              only: pre-prod
+```
+
+## Workspaces
+
+Workflows that include jobs running on multiple branches may require artifacts to be shared using workspaces. To persist an artifact from a job and make it available to other jobs, configure the job to use the `persist_to_workspace` key where the value is a directory inside the project’s working directory. Artifacts of the job will be saved to this directory until the job is rerun and new artifacts are created.
```

Comment:
Workspaces are generally useful for workflows not just when branch filters are involved. E.g. Scala projects typically require lots of CPU for compilation but tests can parallelise across containers well. You could use a larger container for a `build` job and then save the compiled artifacts into the workflow's workspace so that the test containers can use the compiled Scala from the `build` job.